File: BackEnd/DatPhongOnline/Data/Entities/Booking.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace DatPhongOnline.Data.Entities
{
    public class Booking
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public User User { get; set; }
        public DateTime CheckInDate { get; set; }
        public DateTime CheckOutDate { get; set; }
        public int Adults { get; set; }
        public int Children { get; set; }
        public int TotalDays { get; set; }
        [Column(TypeName = "decimal(18,2)")]
        public decimal TotalPrice { get; set; }
        public BookingStatus Status { get; set; }
        public DateTime Createdat { get; set; } = DateTime.UtcNow;

        public ICollection<BookingDetail> BookingDetails { get; set; }
        
    }
}

File: FrontEnd/datphongonline/types/booking.ts
export interface BookingDetail {
  roomId: number;
  pricePerNight: number;
}

export interface BookingRequest {
  checkInDate: string;
  checkOutDate: string;
  adults: number;
  children: number;
  roomIds: number[];
}

export interface SearchRoomRequest {
  checkIn: string;
  checkOut: string;
  adults: number;
  children: number;
}

export enum BOOKING_STATUS {
  PENDING = 0,
  CONFIRM = 1,
  COMPLETE = 2,
  CANCEL = 3,
}

export interface BookingDetail {
  roomId: number;
  roomName: string;
  address: string;
  basePrice: number;
  pricePerNight: number;
}

export interface BookingResponse {
  id: number;
  userId: number;
  checkInDate: string;
  checkOutDate: string;
  adults: number;
  children: number;
  totalDays: number;
  totalPrice: number;
  status: BOOKING_STATUS; // ✅ NUMBER
  bookingDetails: BookingDetail[];
}
// Alias for backward compatibility
export type BookingReponse = BookingResponse;
